ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -88,7 +88,8 @@ else if (desc.getPartitionSpec() != null) {
     CompactionRequest compactionRequest = new CompactionRequest(table.getDbName(), table.getTableName(),
         compactionTypeStr2ThriftType(desc.getCompactionType()));
 
-    compactionRequest.setPoolName(desc.getPoolName());
+    compactionRequest.setPoolName(desc.getPoolName() != null ? desc.getPoolName() : 
```

Comment:
could we use `ObjectUtils.defaultIfNull`?

Reply:
Done